sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -107,6 +107,8 @@ public virtual async Task<Response<ConfigurationSetting>> AddAsync(Configuration
                     case 200:
                     case 201:
                         return await CreateResponseAsync(response, cancellationToken).ConfigureAwait(false);
+                    case 412:
+                        throw await response.CreateRequestFailedExceptionAsync("Setting was already present.").ConfigureAwait(false);
```

Comment:
Should we check to see if the user did pass in an 'match' condition as part of their request before we assume that's why 412 was returned? Probably just gold plating but wanted to ask.